src/bin/pgcopydb/copydb.c

```diff
@@ -679,23 +679,13 @@ copydb_init_specs(CopyDataSpec *specs,
 
 		.sourceTableHashByOid = NULL
 	};
-
-	/* initialize the connection strings */
-	if (!IS_EMPTY_STRING_BUFFER(options->source_pguri))
+	if (tmpCopySpecs.source_pguri == NULL || tmpCopySpecs.target_pguri == NULL ||
+		tmpCopySpecs.sourceSnapshot.pguri == NULL)
 	{
-		strlcpy(tmpCopySpecs.source_pguri,
-				options->source_pguri,
-				MAXCONNINFO);
-
-		strlcpy(tmpCopySpecs.sourceSnapshot.pguri,
-				options->source_pguri,
-				MAXCONNINFO);
+		log_error(ALLOCATION_FAILED_ERROR);
+		return false;
```

Comment:
I wonder if testing against NULL for allocation failure is the right thing to do in this part of the code. The allocation failure happened quite a long time ago already. Maybe we could use `pg_strdup` each place you've been calling `strdup` in that PR. The trouble with `pg_strdup` is that it does `exit(1)` when `strdup` fails, but in that context when parsing the command line options that's not a big problem.